app/src/main/java/com/example/imageEditor/ui/home/adapter/CollectionAdapter.kt

```diff
@@ -0,0 +1,114 @@
+
+class CollectionAdapter : ListAdapter<CollectionModel, CollectionAdapter.ViewHolder>(
+    object : DiffUtil.ItemCallback<CollectionModel>() {
+        override fun areItemsTheSame(
+            oldItem: CollectionModel,
+            newItem: CollectionModel,
+        ): Boolean {
+            return oldItem.id == newItem.id
+        }
+
+        override fun areContentsTheSame(
+            oldItem: CollectionModel,
+            newItem: CollectionModel,
+        ): Boolean {
+            return oldItem.id == newItem.id
+        }
+    },
+) {
+    private val childRecyclerViewState = SparseIntArray()
+
+    class ViewHolder(private val binding: ItemCollectionBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bindView(
+            item: CollectionModel,
+            childRecyclerViewState: SparseIntArray,
+            index: Int,
+        ) {
+            binding.imgUser.displayImage(item.user.profileImage.small)
+            binding.tvUserName.text = item.user.username
+            binding.tvLocation.text = item.user.location
+
+            val adapter = ImageAdapter(binding.root.context, item.previewPhotos)
+            binding.recycleViewImg.adapter = adapter
+
+            binding.tvLikes.text =
+                binding.root.context.getString(
+                    R.string.liked_by_others,
+                    item.coverPhoto.likes.toString(),
+                )
+
+            item.description?.let {
+                val descriptionHtml = "<b>${item.user.username}</b> ${item.description}"
+                binding.tvDescription.text =
+                    HtmlCompat.fromHtml(descriptionHtml, HtmlCompat.FROM_HTML_MODE_LEGACY)
+            }
+
+            binding.recycleViewImg.setHasFixedSize(true)
+
+            if (childRecyclerViewState.containsKey(index)) {
+                val position = childRecyclerViewState[index]
+                binding.recycleViewImg.scrollToPosition(position)
+            }
+            if (binding.recycleViewImg.onFlingListener == null) {
+                val snapHelper = PagerSnapHelper()
+                snapHelper.attachToRecyclerView(binding.recycleViewImg)
+            }
+            binding.recycleViewImg.clearOnScrollListeners()
+
+            binding.recycleViewImg.addOnScrollListener(
+                object : RecyclerView.OnScrollListener() {
+                    override fun onScrolled(
+                        recyclerView: RecyclerView,
+                        dx: Int,
+                        dy: Int,
+                    ) {
+                        super.onScrolled(recyclerView, dx, dy)
+                        val layoutManager = recyclerView.layoutManager
+                        if (layoutManager is LinearLayoutManager) {
+                            childRecyclerViewState[index] =
+                                layoutManager.findLastVisibleItemPosition()
+                        }
+                    }
+                },
```

Comment:
```suggestion
                object : RecyclerView.OnScrollListener() {
                    override fun onScrolled(
                        recyclerView: RecyclerView,
                        dx: Int,
                        dy: Int,
                    ) {
                        super.onScrolled(recyclerView, dx, dy)
                        val layoutManager = recyclerView.layoutManager
                        if (layoutManager is LinearLayoutManager) {
                            childRecyclerViewState[index] =
                                layoutManager.findLastVisibleItemPosition()
                        }
                    }
                },
```
Move đoạn này khai báo thành 1 biến thuộc class ``ViewHolder`` và sử dụng để code dễ đọc hơn kiểu ntn
```kotlin
binding.recycleViewImg.addOnScrollListener(mOnScrollListener)
```